File: movie-infrastructures/storage/src/main/java/com/movie/storage/movie/entity/TheaterEntity.java
package com.movie.storage.movie.entity;

import com.movie.storage.BaseEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Objects;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "theater")
public class TheaterEntity extends BaseEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "BIGINT UNSIGNED")
    private Long id;

    @Column(columnDefinition = "VARCHAR(100) NOT NULL COMMENT '극장명'")
    private String name;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof TheaterEntity theaterEntity)) return false;
        return id != null && id.equals(theaterEntity.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

}

File: movie/infra/src/main/kotlin/com/example/movie/persistence/theater/entity/TheaterEntity.kt
package com.example.movie.persistence.theater.entity

import com.example.movie.domain.theater.model.Theater
import com.example.movie.persistence.common.BaseEntity
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "theater")
class TheaterEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "theater_id", columnDefinition = "INT UNSIGNED")
    val id: Long = 0,

    val name: String,
) : BaseEntity() {
    fun toDomain(): Theater {
        return Theater(
            id = id,
            name = name,
            createdBy = createdBy,
            createdAt = createdAt,
            updatedBy = updatedBy,
            updatedAt = updatedAt
        )
    }
}